cdap-ui/app/directives/timeline/timeline-ctrl.js

```diff
@@ -24,7 +26,63 @@ function TimelineController ($scope, LogViewerStore, LOGVIEWERSTORE_ACTIONS) {
       }
     });
   };
+
+  var pollPromise = null;
+
+  var apiSettings = {
+    metric : {
+      context: `namespace.${this.namespaceId}.app.${this.appId}.flow.${this.programId}.run.${this.runId}`,
+      names: ['system.app.log.error', 'system.app.log.warn', 'system.app.log.info', 'system.app.log.debug'],
+      startTime : '',
+      endTime : '',
+      resolution: '1m'
+    }
+  };
+
+  function pollForMetadata() {
+    pollPromise = dataSrc.poll({
+      _cdapPath: '/metrics/query',
+      method: 'POST',
+      body: MyMetricsQueryHelper.constructQuery(
+        'qid',
+        MyMetricsQueryHelper.contextToTags(apiSettings.metric.context),
+        apiSettings.metric
+      )
+    },
+    function (res) {
+      $scope.metadata = res;
+      $scope.sliderBarPositionRefresh = LogViewerStore.getState().startTime;
+      $scope.initialize();
+      if (res.status === 'KILLED') {
```

Comment:
status KILLED is not the only status where polling needs to stop. There are COMPLETED and FAILED